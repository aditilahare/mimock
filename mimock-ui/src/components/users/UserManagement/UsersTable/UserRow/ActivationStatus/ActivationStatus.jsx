import React from 'react';
import { PropTypes } from 'prop-types';
import {
	Indicator,
	GreenBlob,
	RedBlob,
	StatusLabel,
	ActivationStatusWrapper,
} from './ActivationStatus.style';

function ActivationStatus({ status }) {
	const statusIndicator = () => {
		if (status) {
			return <GreenBlob />;
		}

		return <RedBlob />;
	};
	const statusLabel = status ? 'Active' : 'Inactive';

	return (
		<ActivationStatusWrapper data-testid='user-activation-status'>
			<Indicator data-testid={`status-indicator-${status}`}>
				{statusIndicator()}
			</Indicator>
			<StatusLabel data-testid={`status-label-${status}`}>
				{statusLabel}
			</StatusLabel>
		</ActivationStatusWrapper>
	);
}

ActivationStatus.propTypes = {
	status: PropTypes.bool.isRequired,
};

export default ActivationStatus;
