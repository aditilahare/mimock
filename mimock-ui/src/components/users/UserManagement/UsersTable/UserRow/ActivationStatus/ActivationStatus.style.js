import tw from 'tailwind-styled-components';

export const ActivationStatusWrapper = tw.div`
    flex
    flex-row
    justify-between
    items-center
    align-middle
    gap-2
    mx-auto
`;

export const Indicator = tw.div`
    w-1
`;

export const Blob = tw.div`
    rounded-full
    mx-auto
    block
    p-2
    text-center
`;

export const GreenBlob = tw(Blob)`
    bg-green-500
`;

export const RedBlob = tw(Blob)`
    bg-red-500
`;

export const StatusLabel = tw.div`
    w-24
    font-sans
    font-semibold
    text-center
    text-gray-700
`;
