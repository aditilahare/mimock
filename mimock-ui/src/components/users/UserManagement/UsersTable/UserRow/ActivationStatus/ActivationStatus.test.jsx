import React from 'react';
import { render, act } from '@testing-library/react';
import ActivationStatus from './ActivationStatus';

describe('ActivationStatus', () => {
	it('should render ActivationStatus when status is true', async () => {
		let tree;
		await act(async () => {
			tree = await render(<ActivationStatus status={true} />);
		});

		const { container, getByTestId } = tree;

		expect(getByTestId('user-activation-status')).toBeInTheDocument();

		expect(getByTestId('status-indicator-true')).toBeInTheDocument();
		expect(getByTestId('status-label-true')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	it('should render ActivationStatus when status is false', async () => {
		let tree;
		await act(async () => {
			tree = await render(<ActivationStatus status={false} />);
		});

		const { container, getByTestId } = tree;

		expect(getByTestId('user-activation-status')).toBeInTheDocument();

		expect(getByTestId('status-indicator-false')).toBeInTheDocument();
		expect(getByTestId('status-label-false')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
