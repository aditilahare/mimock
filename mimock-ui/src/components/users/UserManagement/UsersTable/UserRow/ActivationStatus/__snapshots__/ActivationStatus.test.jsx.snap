// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivationStatus should render ActivationStatus when status is false 1`] = `
<div>
  <div
    class="flex flex-row justify-between items-center align-middle gap-2 mx-auto"
    data-testid="user-activation-status"
  >
    <div
      class="w-1"
      data-testid="status-indicator-false"
    >
      <div
        class="rounded-full mx-auto block p-2 text-center bg-red-500"
      />
    </div>
    <div
      class="w-24 font-sans font-semibold text-center text-gray-700"
      data-testid="status-label-false"
    >
      Inactive
    </div>
  </div>
</div>
`;

exports[`ActivationStatus should render ActivationStatus when status is true 1`] = `
<div>
  <div
    class="flex flex-row justify-between items-center align-middle gap-2 mx-auto"
    data-testid="user-activation-status"
  >
    <div
      class="w-1"
      data-testid="status-indicator-true"
    >
      <div
        class="rounded-full mx-auto block p-2 text-center bg-green-500"
      />
    </div>
    <div
      class="w-24 font-sans font-semibold text-center text-gray-700"
      data-testid="status-label-true"
    >
      Active
    </div>
  </div>
</div>
`;
